### PR TITLE
Fix open Contact tab from "customers only" message

### LINKED ISSUE #11784: Contact tab does not open on click on link that changes anchor
## Description

When shop is "Visible to registered customers only" we show a message that
```
Only approved customers can access this shop.
If you're already an approved customer, login to proceed.

Want to start shopping here? Please [contact](http://localhost:3000/fredo-s-farm-hub/shop#contact) Fredo's Farm Hub and ask about joining.
```
Screenshot 2023-11-08 at 13 00 28

Even we update anchor to the new one `#contact`>`#contact_panel` it does not open the tab.
Only in case user refresh the page http://localhost:3000/fredo-s-farm-hub/shop#/contact_panel the Contact tab will be shown

## Expected Behavior

When use click on Contact, than Contact tab should be opened

## Actual Behaviour

It should trigger the event that will open the tab

## Steps to Reproduce

1. Got to enterprise settings, eg local link is http://localhost:3000/admin/enterprises/fred-s-farm/edit#/shop_preferences_panel
2. Check " VISIBLE TO REGISTERED CUSTOMERS ONLY"
3. Open shopfront eg local link is http://localhost:3000/fredo-s-farm-hub/shop
4. Click on Contact link

## Your Environment

* Version used:
* Browser name and version:
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by duleorlovic**
#### What? Why?

We need stimulus attributes that will activate Contact tab

- Closes #11784 

#### What should we test?

- Got to enterprise settings, eg local link is http://localhost:3000/admin/enterprises/fred-s-farm/edit#/shop_preferences_panel
- Check " VISIBLE TO REGISTERED CUSTOMERS ONLY"
- Open shopfront eg local link is http://localhost:3000/fredo-s-farm-hub/shop
- Click on Contact link

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [x] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [ ] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by rioug — APPROVED** [review]
Nice one ! Thanks for your help.
I noticed you merged master into your branch, we prefer rebasing the dev branch on master to keep the git history nice an clean.

**Review by dacook — APPROVED** [review]
Thank you!

That code seems quite wordy, but I see that it's the same method as used elsewhere. Hopefully one day we might be able to simplify it..

**Comment by duleorlovic** [comment]
> Nice one ! Thanks for your help.
> I noticed you merged master into your branch, we prefer rebasing the dev branch on master to keep the git history nice an clean.

Sorry, I have clicked on the github page that I want to rebase on master... Now I have rebased locally and force pushed, so now there is only one commit

**Review by rioug — APPROVED** [review]

**Comment by duleorlovic** [comment]
> Thank you!
> 
> That code seems quite wordy, but I see that it's the same method as used elsewhere. Hopefully one day we might be able to simplify it..

@dacook you are right. We should not wait for refactoring. I added a another commit that solves this [comment](https://github.com/openfoodfoundation/openfoodnetwork/issues/11784#issuecomment-1809544095) , mainly we do not need to use two methods, just `data-action="tabs-and-panels#activate"`.

Removed shop-tabs controllers since we can listen on `"data-action": "orderCycleSelected@window->tabs-and-panels#activateDefaultPanel"`
    
 Added test for cases:
    * activate by clicking on tab
    * activateDefaultPanel on orderCycleSelected event
    * activateFromWindowLocationOrDefaultPanelTarget to activate tab based on achor in URL

@dacook @rioug Please review again

File: spec/javascripts/stimulus/tabs_and_panels_controller_test.js [hunk 1]
```diff
@@ -11,122 +11,143 @@ describe('TabsAndPanelsController', () => {
...
+      expect(booPanel.style.display).toBe("block")
+    })
+  })
+
+  describe("when non valid anchor is specified in the url", () => {
+    const oldWindowLocation = window.location
+    beforeAll(() => {
+      Object.defineProperty(window, "location", {
+        value: new URL("http://example.com/#non_valid_panel"),
+        configurable: true,
```
Comment (rioug) [inline spec/javascripts/stimulus/tabs_and_panels_controller_test.js:128]: It's a lot cleaner than the previous solution !

**Review by rioug — APPROVED** [review]
Looks great ! thanks for cleaning that up :pray:

**Comment by mkllnk** [comment]
There are a couple of specs failing.

**Comment by sigmundpetersen** [comment]
@duleorlovic did you want to fix the failing specs?
Then this can move forward 👍

File: spec/javascripts/stimulus/tabs_and_panels_controller_test.js (hunk 1 above)
Comment (duleorlovic) [reply spec/javascripts/stimulus/tabs_and_panels_controller_test.js:128]: Yea, no need to use mock

**Comment by duleorlovic** [comment]
> There are a couple of specs failing.

Sorry for late reply. I was thinking that there were som Flaky test, but I had to update some test case also.
I used `data-test=link_for_enterprise_fees` instead of `id=enterprise_fees` so it is obvious that we are using for `expect(page).to have_selector "[data-test=link_for_enterprise_fees]"`

**Review by mkllnk — APPROVED** [review]
Great, thank you!

**Review by rioug — APPROVED** [review]

**Comment by drummer83** [comment]
Hi @duleorlovic,
Thank you for this contribution! I have tested this PR on our staging server and it seems to work great! :muscle: 

I have tested that
- with the 'registered users only' option **enabled**
  - guest users see the contact message and clicking the contact link works well, as well as the login link,
  - logged-in users (no customers yet) see the contact message and clicking the link works well,
  - logged-in customers see the products right away,
- with the 'registered users only' option **disabled**
  - guest users see the products right away,
  - logged-in users (no customers yet) see the products right away,
  - logged-in customers see the products right away.

Here is a short video (covering only parts of the tests):

https://github.com/openfoodfoundation/openfoodnetwork/assets/1790176/1b173c5d-4f0d-406b-83a1-cd833a17b266

I think this one ready to go! :partying_face: 
I'll go ahead and merge this one! :rocket: 
Thanks again!

**Pull request merged**